tests/integration/utils2.go

```diff
@@ -739,33 +751,6 @@ func restartNodes(
 		}
 	}
 
-	// The index of the action after the last wait action before the current restart action.
-	// We wish to resume the wait clock from this point onwards.
-	waitGroupStartIndex := 0
-actionLoop:
-	for i := actionIndex; i >= 0; i-- {
-		switch s.testCase.Actions[i].(type) {
-		case WaitForSync:
-			// +1 as we do not wish to resume from the wait itself, but the next action
-			// following it. This may be the current restart action.
-			waitGroupStartIndex = i + 1
-			break actionLoop
-		}
-	}
-
-	for _, tc := range s.testCase.Actions {
-		switch action := tc.(type) {
-		case ConnectPeers:
-			// Give the nodes a chance to connect to each other and learn about each other's subscribed topics.
-			time.Sleep(100 * time.Millisecond)
```

Comment:
praise: I forgot we were currently relying on a sleep - thank you very much for removing this!